### ISSUE TITLE
Parsing with H and zero hour fails

### ISSUE THREAD

**Issue by lukebaker** [taylorhakes/fecha#87]
Tested on node 14.15.1:

```
> fecha.parse("4/18/2021 01:41:42", "M/D/YYYY HH:mm:ss")
2021-04-18T05:41:42.000Z
> fecha.parse("4/18/2021 00:41:42", "M/D/YYYY HH:mm:ss")
2021-04-18T04:41:42.000Z
> fecha.parse("4/18/2021 1:41:42", "M/D/YYYY H:mm:ss")
2021-04-18T05:41:42.000Z
> fecha.parse("4/18/2021 0:41:42", "M/D/YYYY H:mm:ss")
null
```

The first three parse as expected, but the last one fails to parse. I would have expected it to be: `2021-04-18T04:41:42.000Z`. Looking at the code, it appears the `H` when parsing uses the `twoDigitsOptional` regular expression, which only matches if it starts with a digit 1-9, whereas the documentation suggestions that `H` can parse `0 1 ... 22 23`.

**Comment by taylorhakes** [comment]
Thanks for catching this @lukebaker . If you create a PR I will happily merge it

**Comment by taylorhakes** [comment]
Fixed issue in 4.2.2
